addons/point_of_sale/static/src/js/models.js

```diff
@@ -2009,7 +2009,7 @@ exports.Orderline = Backbone.Model.extend({
         }
 
         // just like in sale.order changing the quantity will recompute the unit price
-        if(! keep_price && ! this.price_manually_set){
+        if(! keep_price && ! this.price_manually_set && ! this.sale_order_line_id){
```

Comment:
I think it would be better to figure out why `this.price_manually_set` isn't true.

Reply:
```suggestion
        if (!keep_price && !this.price_manually_set && !this.sale_order_line_id){
```
same as above